src/base/bittorrent/cachestatus.cpp

```diff
@@ -50,7 +50,11 @@ qreal CacheStatus::readRatio() const
 
 int CacheStatus::jobQueueLength() const
 {
+#if LIBTORRENT_VERSION_NUM < 10100
     return m_nativeStatus.job_queue_length;
+#else
+    return m_nativeStatus.queued_jobs;
```

Comment:
the session_stats doesn't necessarily provide averages. It provide counters and timestamps. from that you can average the numbers over any time window. For disk I/O related counters and gauges, you'll find them here: https://github.com/arvidn/libtorrent/blob/RC_1_1/src/session_stats.cpp#L319

disk_blocks_in_use may be what you're looking for, for example